busybox/busybox.py

```diff
@@ -9,39 +9,63 @@
 
 from trunner.ctx import TestContext
 from trunner.dut import Dut
-from trunner.types import TestResult, TestSubResult, Status
+from trunner.types import TestResult, Status
+
+# Example of parsed outputs:
+#
+#   ---Testcase test-name-1 starting---
+#   test-msg-line-1
+#   test-msg-line-2
+#   FAIL: test-name-1
+#   fail-msg-line-1
+#   fail-msg-line-2
+#   ---Testcase test-name-2 starting---
+#   test-msg-line-1
+#   test-msg-line-2
+#   PASS: test-name-2
+#   ---Testcase test-name-3 starting---
+#   SKIPPED: test-name-3
+#
 
 
 def harness(dut: Dut, ctx: TestContext, result: TestResult):
     subresult = None
     msg = []
     test_status = Status.OK
 
+    START = r"---Testcase (?P<name>.*?)starting---\r+\n"
```

Comment:
should probably be 
```suggestion
    START = r"---Testcase '(?P<name>.+?)' starting---\r+\n"
```

use regexp101.com to test the regexes against sample output provided above